### PR TITLE
Fixes for vfile.cpp

### PR DESCRIPTION
**Pull request by z33ky**

### CODE REVIEW

File: source/SH3/arc/vfile.cpp [hunk 1]
```diff
@@ -55,15 +60,22 @@ void sh3_arc_vfile::Seek(long pos, std::ios_base::seekdir origin)
 {
     if(origin == std::ios_base::beg)
     {
-        fpos = pos;
+        fpos = static_cast<decltype(fpos)>(pos);
     }
-    else if(std::ios_base::cur)
+    else if(origin == std::ios_base::cur)
     {
-        fpos += pos;
```
Comment (z33ky) [inline source/SH3/arc/vfile.cpp:62]: This line might be fine with C++'s convertion rules and unsigned integer overflow. Need to investigate...

Comment (z33ky) [reply source/SH3/arc/vfile.cpp:62]: See the following amended change...

Comment (Quaker762) [reply source/SH3/arc/vfile.cpp:62]: Alright, do you want me to merge or look this up first?

File: source/SH3/arc/vfile.cpp [hunk 2]
```diff
@@ -23,13 +23,18 @@ bool sh3_arc_vfile::Open(sh3_arc& mft, const std::string& fname)
         it (so we know how large it is without probing) though most headers contain the size of the
         full file
     */
-    if((fsize = mft.LoadFile(fname, buffer)) == ARC_FILE_NOT_FOUND)
+    int size = mft.LoadFile(filename, buffer);
```
Comment (Quaker762) [inline source/SH3/arc/vfile.cpp:26]: Haha, can you tell where I learned C (*cough*K&R*cough*)

**Review by Quaker762 — APPROVED** [review]

File: source/SH3/arc/vfile.cpp (hunk 2 above)
Comment (z33ky) [reply source/SH3/arc/vfile.cpp:26]: It's a perfectly fine technique, however clang complains about some implicit casts, which I avoid by buffering the value in a local `int`.

File: source/SH3/arc/vfile.cpp (hunk 1 above)
Comment (z33ky) [reply source/SH3/arc/vfile.cpp:62]: The comment below gives a rough explanation why it's fine. Feel free to verify that it's true though.

File: source/SH3/arc/vfile.cpp (hunk 2 above)
Comment (Quaker762) [reply source/SH3/arc/vfile.cpp:26]: Ah, I see. g++ doesn't seem to complain about this. Maybe I need to enable more warnings in Code::Blocks

**Pull request merged**